src/ChatHistoryItem.tsx

```diff
@@ -38,20 +38,23 @@ export const ChatHistoryItem = memo(
       setIsMenuOpen((prev) => !prev);
     };
 
-    const handleSaveTitle = useCallback(() => {
-      const trimmedTitle = editInputValue.trim();
```

Comment:
This was getting defined at the time the function was created, and we should always be passing in the current state value of the input at the time this function is called
- on enter key press
- on outside click